pkg/internal/token/provider.go

```diff
@@ -22,7 +23,7 @@ type CredentialProvider interface {
 	Name() string
 }
 
-func NewAzIdentityCredential(record azidentity.AuthenticationRecord, o *Options) (CredentialProvider, error) {
+func NewAzIdentityCredential(record azidentity.AuthenticationRecord, cache cache.ExportReplace, o *Options) (CredentialProvider, error) {
```

Comment:
I'll let @weinong decide, but another way could be that we either merge both caches into a new struct or directly in the Options maybe ?